### DISCUSSION TITLE
É possível fazer pesquisas por CNAE, endereço ou outros filtros

### DISCUSSION THREAD

**Discussion by carloswagnerbg** [rictom/rede-cnpj#17] [General]
Primeiramente parabéns pelo projeto!

Gostaria de saber se é possível filtrar as empresas por CNAE, endereço ou outros filtros. Caso não seja possível, tem alguma dica para conseguir fazer isso?

**Comment by rictom** [comment]
Pra fazer uma consulta desse tipo, sugiro vc trabalhar direto com a base cnpj.db no [DBrowser](https://sqlitebrowser.org/)
Por exemplo, esse é um comando SQL para filtrar por natureza juridica
https://github.com/rictom/cnpj-sqlite/blob/main/exemplos/sql_consulta_dados_empresas_socios_por_natureza.sql
Vc teria que adaptar e trocar os nomes dos parâmetros dessa consulta pelo CNAE.

Na base cnpj.db, da mesma maneira, vc consegue filtrar empresas por municipio ou UF. Endereço mesmo, como nome de rua, pode dar resultado ruim, porque a forma que o cadastro não é preenchido de maneira uniforme. Mas poderia filtrar por CEP.

**Reply by carloswagnerbg** [reply]
Obrigado, deu certo!
